src/H5FDdevelop.h

```diff
@@ -161,80 +185,163 @@
+    /**< Frees the resources for this driver */
+
     haddr_t (*get_eoa)(const H5FD_t *file, H5FD_mem_t type);
+    /**< Gets the address of first byte past the addressed space */
+
     herr_t (*set_eoa)(H5FD_t *file, H5FD_mem_t type, haddr_t addr);
+    /**< Sets the end-of-address marker for the file */
+
     haddr_t (*get_eof)(const H5FD_t *file, H5FD_mem_t type);
+    /**< Gets the address of first byte past the file-end */
+
     herr_t (*get_handle)(H5FD_t *file, hid_t fapl, void **file_handle);
+    /**< Returns the file handle of this file driver */
+
     herr_t (*read)(H5FD_t *file, H5FD_mem_t type, hid_t dxpl, haddr_t addr, size_t size, void *buffer);
+    /**< Reads the specified number of bytes of data from the file beginning at the specified
+     * address into the provided buffer, according to the specified data transfer properties  */
+
     herr_t (*write)(H5FD_t *file, H5FD_mem_t type, hid_t dxpl, haddr_t addr, size_t size, const void *buffer);
+    /**< Writes the specified number of bytes of data to the file beginning at the specified
+     * address from the provided buffer, according to the specified data transfer properties  */
+
     herr_t (*read_vector)(H5FD_t *file, hid_t dxpl, uint32_t count, H5FD_mem_t types[], haddr_t addrs[],
                           size_t sizes[], void *bufs[]);
+    /**< Reads the specified length of data from the file into the provided array */
+
     herr_t (*write_vector)(H5FD_t *file, hid_t dxpl, uint32_t count, H5FD_mem_t types[], haddr_t addrs[],
                            size_t sizes[], const void *bufs[]);
+    /**< Writes the specified length of data in the provided array to the file at the specified offsets */
+
     herr_t (*read_selection)(H5FD_t *file, H5FD_mem_t type, hid_t dxpl_id, size_t count, hid_t mem_spaces[],
                              hid_t file_spaces[], haddr_t offsets[], size_t element_sizes[],
                              void *bufs[] /*out*/);
+    /**< */
+
     herr_t (*write_selection)(H5FD_t *file, H5FD_mem_t type, hid_t dxpl_id, size_t count, hid_t mem_spaces[],
                               hid_t file_spaces[], haddr_t offsets[], size_t element_sizes[],
                               const void *bufs[] /*in*/);
+    /**< */
+
     herr_t (*flush)(H5FD_t *file, hid_t dxpl_id, hbool_t closing);
+    /**< Flushes all data to disk */
+
     herr_t (*truncate)(H5FD_t *file, hid_t dxpl_id, hbool_t closing);
+    /**< Truncates a file */
+
     herr_t (*lock)(H5FD_t *file, hbool_t rw);
+    /**< Places an advisory lock on a file */
+
     herr_t (*unlock)(H5FD_t *file);
+    /**< Removes the existing lock on a file */
+
     herr_t (*del)(const char *name, hid_t fapl);
+    /**< Deletes a file */
+
     herr_t (*ctl)(H5FD_t *file, uint64_t op_code, uint64_t flags, const void *input, void **output);
+    /**< Performs a CTL operation */
+
     H5FD_mem_t fl_map[H5FD_MEM_NTYPES];
+    /**< Free-list map */
```

Comment:
Shouldn't these be before the struct member?

Reply:
These are inline, so should be after, I believe.